drivers/w5500/w5500.c

```diff
@@ -0,0 +1,446 @@
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "assert.h"
+#include "iolist.h"
+#include "net/ethernet.h"
+#include "net/eui_provider.h"
+#include "net/netdev/eth.h"
+#include "net/ethernet/hdr.h"
+#include "ztimer.h"
+
+#include "w5500.h"
+#include "w5500_regs.h"
+
+#define ENABLE_DEBUG        0
+#include "debug.h"
+#include "log.h"
+
+/* Socket definitions. */
+#define BSB_SOCKET_BASE         (0x08)      /**< Base for Socket n registers. */
+#define RXBUF_16KB_TO_S0        (0x10)      /**< Receive memory size: 16 kB. */
+#define TXBUF_16KB_TO_S0        (0x10)      /**< Transmit memory size: 16 kB. */
+
+static const netdev_driver_t netdev_driver_w5500;
+
+static uint8_t read_register(w5500_t *dev, uint16_t reg)
+{
+    uint16_t address = reg & 0x07ff;
+    uint8_t command = (uint8_t)((reg & 0xf800) >> 8u) | CMD_READ;
+
+    spi_transfer_u16_be(dev->p.spi, dev->p.cs, true, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    return spi_transfer_byte(dev->p.spi, dev->p.cs, false, 0u);
+}
+
+static void write_register(w5500_t *dev, uint16_t reg, uint8_t data)
+{
+    uint16_t address = reg & 0x07ff;
+    uint8_t command = (uint8_t)((reg & 0xf800) >> 8u) | CMD_WRITE;
+
+    spi_transfer_u16_be(dev->p.spi, dev->p.cs, true, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, false, data);
+}
+
+static uint16_t read_addr(w5500_t *dev, uint16_t addr_high, uint16_t addr_low)
+{
+    uint16_t res = (read_register(dev, addr_high) << 8);
+
+    res |= read_register(dev, addr_low);
+    return res;
+}
+
+static void write_address(w5500_t *dev,
+                  uint16_t addr_high, uint16_t addr_low, uint16_t val)
+{
+    write_register(dev, addr_high, (uint8_t)(val >> 8));
+    write_register(dev, addr_low, (uint8_t)(val & 0xff));
+}
+
+static void read_chunk(w5500_t *dev, uint16_t addr, uint8_t *data, size_t len)
+{
+    uint16_t address = addr & 0x07ff;
+    uint8_t command = (uint8_t)((addr & 0xf800) >> 8u) | CMD_READ;
+
+    spi_transfer_u16_be(dev->p.spi, dev->p.cs, true, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    spi_transfer_bytes(dev->p.spi, dev->p.cs, false, NULL, data, len);
+}
+
+static void write_chunk(w5500_t *dev, uint16_t addr, uint8_t *data, size_t len)
+{
+    uint16_t address = addr & 0x07ff;
+    uint8_t command = (uint8_t)((addr & 0xf800) >> 8u) | CMD_WRITE;
+
+    spi_transfer_u16_be(dev->p.spi, dev->p.cs, true, address);
+    spi_transfer_byte(dev->p.spi, dev->p.cs, true, command);
+    spi_transfer_bytes(dev->p.spi, dev->p.cs, false, data, NULL, len);
+}
+
+static void extint(void *arg)
+{
+    w5500_t *dev = (w5500_t *)arg;
+
+    netdev_trigger_event_isr(&dev->netdev);
+    if (!gpio_is_valid(dev->p.irq)) {
+        /* restart timer if we are polling */
+        (void)ztimer_set(ZTIMER_MSEC, &dev->timerInstance, dev->p.polling_interval_ms);
```

Comment:
```suggestion
        ztimer_set(ZTIMER_MSEC, &dev->timerInstance, dev->p.polling_interval_ms);
```

Reply:
Hi, I have added your comments and squashed everything.

Reply:
This change is not in though.

Reply:
Gnah! But now! :-)